### PR TITLE
feat(vue-demo): message for increasing product quantity above stock

### LINKED ISSUE #1167: [BUG] Error message for increasing product quantity above stock
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

When we try to increase the quantity of a product above the stock level, we receive a message without information about the available stock
 
image

### Expected Behavior

_No response_

### Steps To Reproduce

_No response_

### Environment

```markdown
- OS:
- Node:
- pnpm:
```

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by mdanilowicz**
### Description

- Added `promotions-on-cart-price-zero-error` error translation
- Added `product-stock-reached-empty` error translation
- Fixed add to cart notifications
- Added `useCmsTranslate` composable for replacing placeholders
- Added `useCartErrorParamsResolver` composable for building error message parameters
- Added `getItem` method to `useCart` for getting single item from cart

closes #1167 #782

### CODE REVIEW

File: packages/cms-base/components/SwProductAddToCart.vue [hunk 1]
```diff
@@ -33,9 +49,16 @@ const { addToCart, quantity } = useAddToCart(product);
 
 const addToCartProxy = async () => {
   await addToCart();
-  pushSuccess(
-    `${props.product?.translated.name} ${translations.product.addedToCart}`,
-  );
+  const errors = getErrorsCodes();
+  errors?.forEach((element) => {
+    const { messageKey, params } = useCartErrorParamsResolver(element);
```
Comment (patzick) [inline packages/cms-base/components/SwProductAddToCart.vue:54]: you should not use composable inside any function
somthing like:

```ts
const { resolveErrorParam } = useCartErrors()
```

and here
```ts
 errors?.forEach((element) => {
    const { messageKey, params } = resolveErrorParam(element);
    pushError(cmsT(translations.errors[messageKey], params));
  });
```

File: packages/composables/src/useCart.ts [hunk 2]
```diff
@@ -278,6 +279,10 @@ export function useCartFunction(): UseCartReturn {
     return errors;
   };
 
+  const getItem = (id: string) => {
```
Comment (patzick) [inline packages/composables/src/useCart.ts:282]: I'd skip it from modifying `useCart` api and just add this inside error resolver, we could add this here if there would be more broadly needed

**Review by patzick — CHANGES_REQUESTED** [review]

File: packages/composables/src/useCmsTranslations.ts [hunk 3]
```diff
@@ -3,3 +3,33 @@ import { inject } from "vue";
 export function useCmsTranslations() {
   return inject("cmsTranslations", {});
 }
+
+export function useCmsTranslate() {
```
[the remark this answers is not in the document]
Comment (mdanilowicz) [reply packages/composables/src/useCmsTranslations.ts]: done

File: packages/cms-base/components/SwProductAddToCart.vue (hunk 1 above)
Comment (mdanilowicz) [reply packages/cms-base/components/SwProductAddToCart.vue]: done

File: packages/composables/src/useCart.ts (hunk 2 above)
Comment (mdanilowicz) [reply packages/composables/src/useCart.ts]: done

**Review by patzick — APPROVED** [review]

**Pull request merged**